hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientMetrics.java

```diff
@@ -97,10 +131,26 @@ public void decrPendingContainerOpsMetrics(ContainerProtos.Type type) {
   }
 
   public void addContainerOpsLatency(ContainerProtos.Type type,
-      long latencyMillis) {
+      long latencyMillis, String datanode) {
+    recordLatencyMetricsIfNeeded(type, latencyMillis, datanode);
     containerOpsLatency.get(type).add(latencyMillis);
   }
 
+  private void recordLatencyMetricsIfNeeded(ContainerProtos.Type type, long latencyMillis,
+      String datanode) {
+    if (latencyMsThresholdsOrder.length > 0 && topLatencyMetricsSet.contains(type)) {
```

Comment:
This seems like a lot of computation added in the critical data path.

Reply:
I think we can do a benchmark, and get a CPU Flame Graphs for a Datanode to check whether it is a heavier operation

Reply:
I did a benchmark use command `ozone freon ommg --operation READ_KEY -n 1000000 -t 400  --bucket bucket1 --volume s3v --size 4k  --duration=1000`, the command generated about 25000 QPS read.

The `recordLatencyMetricsIfNeeded` only 0.07% on the CPU time slice
![image](https://github.com/apache/ozone/assets/32928346/09e9812d-0c1c-4bf7-9319-1ac84dc587a8)
And the Flame Graphs have been put to this PR's jira ticket https://issues.apache.org/jira/secure/attachment/13069380/client-readkey-25000-4k-cpu.html